src/AISmart.GAgent.Twitter/Provider/TwitterProvider.cs

```diff
@@ -3,126 +3,201 @@
 using System.Linq;
 using System.Net.Http;
 using System.Net.Http.Headers;
+using System.Security.Cryptography;
 using System.Text;
 using System.Threading.Tasks;
-using AElfScanServer.Worker.Core.Dtos;
+using AISmart.Dto;
 using AISmart.Options;
+using AISmart.Util;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Newtonsoft.Json;
-using Volo.Abp;
 using Volo.Abp.DependencyInjection;
 
 namespace AISmart.Provider;
 
 public interface ITwitterProvider
 {
-    public  Task<List<Tweet>> GetLatestTwittersAsync(string sendUser, string userId, string sinceTweetId);
-
-    public Task<string> PostTwitterAsync(string sendUser, string message);
+    public Task<string> PostTwitterAsync(string message, string accessToken, string accessTokenSecret);
+    public Task<string> ReplyAsync(string message, string tweetId, string accessToken, string accessTokenSecret);
+    public Task<List<Tweet>> GetMentionsAsync();
 }
 
 
 public class TwitterProvider : ITwitterProvider, ISingletonDependency
 {
     private readonly ILogger<ITwitterProvider> _logger;
-    private readonly IOptionsMonitor<TwitterOptions> _telegramOptions;
-    public TwitterProvider(ILogger<ITwitterProvider> logger,IOptionsMonitor<TwitterOptions> telegramOptions)
+    private readonly IOptionsMonitor<TwitterOptions> _twitterOptions;
+    private readonly HttpClient _httpClient;
+    private readonly AESCipher _aesCipher;
+    
+    public TwitterProvider(ILogger<ITwitterProvider> logger, IOptionsMonitor<TwitterOptions> twitterOptions)
     {
         _logger = logger;
-        _telegramOptions = telegramOptions;
+        _twitterOptions = twitterOptions;
+        _httpClient = new HttpClient();
+        string password = twitterOptions.CurrentValue.EncryptionPassword;
+        _aesCipher = new AESCipher(password);
     }
-
-    public async Task<List<Tweet>> GetLatestTwittersAsync(string sendUser, string userId, string sinceTweetId)
+    
+    public async Task<List<Tweet>> GetMentionsAsync()
     {
-        var bearerToken = GetAccount(sendUser);
-        string url = $"https://api.twitter.com/2/tweets"; 
-        int maxResults = 3; 
-        _logger.LogInformation($"twitterToken = {bearerToken}, userId = {userId},sinceTweetId = {sinceTweetId}");
+        var bearerToken = _twitterOptions.CurrentValue.BearerToken;
+        string username = "elonmusk";
```

Comment:
Why is the username hardcoded? We can only get mentions from elonmusk's account?